src/vault.rs

```diff
@@ -135,6 +144,20 @@ impl Client {
         Ok(())
     }
 
+    pub fn kubernetes_auth(&mut self, role: String) -> Result<(), Error> {
+        let mut file = File::open("/var/run/secrets/kubernetes.io/serviceaccount/token").expect("Unable to open");
+        let mut jwt = String::new();
+        file.read_to_string(&mut jwt);
+        // .is_err() {
+        //     return Err(Error::);
+        //     // return Err(String::from("Kubernetes authentication failed"));
+        // };
+        let request = AuthKubernetesRequest { jwt, role };
+        let response: AuthResponseWrapper = self.post(&format!("auth/kubernetes/login"), &request)?;
```

Comment:
No need to use format here, as you're not adding anything to the string, just `self.post("auth/kubernetes/login", &request)?;` should do.